### ISSUE TITLE
Why does the header needs to have unique values?

### ISSUE THREAD

**Issue by NicolasCARPi** [thephpleague/csv#380]
|    Q        |   A
|------------ | ------
| Version     | 9.5.0

### Question

Hello,

First, good job on this library, it's a real pleasure to use and the code is very clean, so thank you to all contributors! :clap: 

I have a question relative to the header when reading a CSV. [The code](https://github.com/thephpleague/csv/blob/master/src/Reader.php#L315) is checking if all values are unique. This leads to issues like this one: https://github.com/elabftw/elabftw/issues/1748

Having columns that have the same value is valid IMHO.

I would like to know if there is a reason for this voluntary limitation, and if yes, which one.

Thank you, have a nice day! :)

~Nico

**Comment by nyamsprod** [comment]
@NicolasCARPi thanks for the kind words. 

About your question yes there's a simple explanation for that. The package treats CSV document as tabular data structure similar to RDBS table as such it enforces RDBS limitations: **you can not have multiple columns with the same name.**

That being said, nothing prevents you from having multiple columns with the same name *but* since records consumed using PHP's own algorithm (`fgetcsv`) you will nevertheless only return the value of the last column. 

TL;DR: the restriction is enforce by the package design and by the language feature there are ways to work around them but for most regular cases which are addressed by this package it is not worth the effort.

**Comment by NicolasCARPi** [comment]
Thank you for the detailed answer @nyamsprod !
